sqlglot/expressions.py

```diff
@@ -248,17 +248,37 @@ def meta(self) -> t.Dict[str, t.Any]:
         return self._meta
 
     def __deepcopy__(self, memo):
-        copy = self.__class__(**deepcopy(self.args))
-        if self.comments is not None:
-            copy.comments = deepcopy(self.comments)
-
-        if self._type is not None:
-            copy._type = self._type.copy()
-
-        if self._meta is not None:
-            copy._meta = deepcopy(self._meta)
-
-        return copy
+        root = self.__class__()
+        stack = [(self, root)]
+        root.parent = self.parent
+
+        while stack:
+            node, copy = stack.pop()
+
+            if node.comments is not None:
+                copy.comments = deepcopy(node.comments)
+            if node._type is not None:
+                copy._type = node._type.copy()
+            if node._meta is not None:
+                copy._meta = deepcopy(node._meta)
+
+            for k, v in node.args.items():
+                if isinstance(v, Expression):
+                    c = v.__class__()
+                    copy.set(k, c)
+                    stack.append((v, c))
+                elif isinstance(v, list):
+                    for e in v:
```

Comment:
FYI the failing test has to do with this branch - the edge case is `[]`. We need to copy that too.